### ISSUE TITLE
Optimize Imports checkbox set on formatter breaks code

### ISSUE THREAD

**Issue by Zteve** [intellij-rust/intellij-rust#2276]
Although I see an issue switching this feature on (and presumably optionally for the formatter), this is the bug that the formatter actually _breaks_ code when this flag is checked.  The flag _should_ be ignored if the function is not yet working properly.  Reformatting should _never_ break working code.

For example, imports in inline, embedded modules (for tests, for example) are moved to the top of the source file and merged (as-is) with the other imports. This means that all the sub-module code breaks.

Rust plugin: 0.2.0.2086; IntelliJ: (Ultimate) 2017.3.4

**Comment by farodin91** [comment]
Do you like to add a code?

**Comment by vsuharnikov** [comment]
@farodin91 

Before the import optimisation:
```rust
use std::convert::From;

fn foo(x: u8) -> u8 {
    x + 1
}

#[cfg(test)]
mod test {
    use std::convert::From;
    use super::foo;

    #[test]
    fn foo_test() {
        let x: u8 = From::from(100);
        assert_eq!(foo(x), x + 1);
    }
}
```

After:
```rust
use std::convert::From;

fn foo(x: u8) -> u8 {
    x + 1
}

use std::convert::From;
use super::foo;

#[cfg(test)]
mod test {
    #[test]
    fn foo_test() {
        let x: u8 = From::from(100);
        assert_eq!(foo(x), x + 1);
    }
}
```

But if you remove the first line:
```rust
use std::convert::From;
```
the test imports weren't changed after the optimisation.
